test/other/embind_tsgen_ignore_1.d.ts

```diff
@@ -31,7 +31,7 @@ export interface ClassHandle {
   delete(): void;
   deleteLater(): this;
   isDeleted(): boolean;
-  clone(): this;
+  clone(): ThisType<this>;
```

Comment:
What does this expression do/mean?

Reply:
IIUC, it's magic marker that changes how the `this` type is bound using the context it was declared instead of where it was called from. This is a workaround for some g3 conflicts, but I'm actually leaning towards abandoning this PR and asking that they don't try to override the clone method.

Reply:
Seems a little magical but LGTM if it fixes the issue!